libretro-common/include/libretro.h

```diff
@@ -976,6 +977,85 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+/* Seek flags from stdio.h */
+enum retro_seek_flag
+{
+  RETRO_SEEK_SET = 0, /* Beginning of file */
+  RETRO_SEEK_CUR = 1, /* Current position of the file pointer */
+  RETRO_SEEK_END = 2, /* End of file - implementation not required */
+};
+
+/* Stat fields */
+struct retro_stat_struct
+{
+  uint32_t deviceId;         /* ID of device containing file */
+  uint64_t size;             /* Total size, in bytes */
+  timespec accessTime;       /* Time of last access */
+  timespec modificationTime; /* Time of last modification */
+  timespec statusTime;       /* Time of last status change */
+  bool isDirectory;          /* The stat url is a directory */
+  bool isSymLink;            /* The stat url is a symbolic link */
+  bool isHidden;             /* The file is hidden */
+};
+
+/* Opaque file handle */
+typedef struct retro_file retro_file;
+
+/* File operations */
+typedef retro_file* (RETRO_CALLCONV *retro_open_file_t)(const char *url, int flags);
+typedef retro_file* (RETRO_CALLCONV *retro_open_file_for_write_t)(const char *url, bool overwrite);
+typedef int64_t (RETRO_CALLCONV *retro_read_file_t)(retro_file *file, uint8_t *buffer, size_t buffer_size);
+typedef int64_t (RETRO_CALLCONV *retro_write_file_t)(retro_file *file, const uint8_t *buffer, size_t buffer_size);
+typedef int64_t (RETRO_CALLCONV *retro_seek_file_t)(retro_file *file, int64_t position, retro_seek_flag whence);
+typedef int64_t (RETRO_CALLCONV *retro_get_file_position_t)(retro_file *file);
+typedef int64_t (RETRO_CALLCONV *retro_get_file_length_t)(retro_file *file);
```

Comment:
size, not length. Not sure about 'position' either, feels weird... but that's what Windows and C use (Linux calls it 'offset', which isn't an improvement). 'Pointer' maybe.

Or maybe ditch the entire concept and add a position argument to read/write. (We also need clear docs on what happens if a write goes outside the file. Maybe we want a truncation function too.)